src/html.rs

```diff
@@ -51,34 +54,81 @@ pub fn simplify(html: &str, lang: &str) -> String {
             }
         }
 
-        for id in to_remove.drain(..) {
-            if let Some(mut node) = document.tree.get_mut(id) {
-                node.detach();
-            }
-        }
+        remove_ids(&mut document, to_remove.drain(..));
     } else {
         warn!("No sections to remove configured for lang {lang:?}");
     }
 
-    // Remove elements with no text that isn't whitespace.
+    // Handle remaining removals in single pass.
 
-    for element in document
+    for el in document
         .root_element()
         .descendants()
         .filter_map(ElementRef::wrap)
     {
-        if element.text().all(|t| t.trim().is_empty()) {
-            to_remove.push(element.id());
+        if is_image(&el) || is_empty_or_whitespace(&el) {
+            to_remove.push(el.id());
         }
     }
+    remove_ids(&mut document, to_remove.drain(..));
+
+    // Final transformations.
 
-    for id in to_remove.drain(..) {
+    let base_url = Url::parse(&format!("https://{}.wikipedia.org/wiki/", &lang)).unwrap();
+    fix_relative_urls(&mut document, base_url);
+
+    document.html()
+}
+
+fn remove_ids(document: &mut Html, ids: impl IntoIterator<Item = NodeId>) {
+    for id in ids {
         if let Some(mut node) = document.tree.get_mut(id) {
             node.detach();
         }
     }
+}
 
-    document.html()
+fn is_empty_or_whitespace(el: &ElementRef) -> bool {
+    el.text().all(|t| t.trim().is_empty())
+}
+
+fn is_image(el: &ElementRef) -> bool {
+    ["img", "picture"].contains(&el.value().name())
+}
+
+/// Convert relative links to absolute links that start with `base`.
+///
+/// In the dumps, links within the same language's wiki are written as relative
+/// links (`./Article_Title`), while cross-wiki links (e.g. `[[:fr:Pomme]]`) are
+/// already expanded to absolute links (`https://fr.wikipedia.org/wiki/Pomme`).
+///
+/// See https://en.wikipedia.org/wiki/Help:Interwiki_linking for more examples of wikitext linking.
+fn fix_relative_urls(document: &mut Html, base: Url) {
+    let href = QualName::new(None, Namespace::from(""), LocalName::from("href"));
+
+    for element in document.tree.values_mut().filter_map(|n| match n {
+        Node::Element(el) if el.name() == "a" => Some(el),
```

Comment:
I don't see any links at all in the existing HTML dumps... Only formatting and headers tags.

Reply:
Yes, the scraper uses [the Extracts API](https://www.mediawiki.org/wiki/Extension:TextExtracts) which strips most of the non-text elements.
Would you prefer we removed the links altogether?

Reply:
Links may be useful in the future, especially if we somehow link wiki articles that are already embedded in the offline maps data. But for now, they can be omitted, I didn't see them on Wiki pages in OM. Did you?

Reply:
No, the current ones don't have any. I'll strip them from the output for now.